pkg/queue/instance.go

```diff
@@ -97,7 +113,18 @@ func (q *queueImpl) pushRetryTask(item *BackoffTask) {
 	q.cond.Signal()
 }
 
+func (q *queueImpl) Closed() <-chan struct{} {
+	return q.closed
+}
+
 func (q *queueImpl) Run(stop <-chan struct{}) {
+	log.Debugf("started queue %s", q.id)
+	defer func() {
+		q.closeOnce.Do(func() {
```

Comment:
maybe remove closeOnce

Reply:
This helps with the goroutine leak test.. the closeOnce helps in scenarios where we `Run` more than once (which we should never do).. but i'd rather the queue be robust to improper usage.